apps/aecore/lib/aecore/miner/worker.ex

```diff
@@ -290,15 +268,56 @@ defmodule Aecore.Miner.Worker do
   defp filter_transactions_by_fee(txs) do
     miners_fee_bytes_per_token = Application.get_env(:aecore, :tx_data)[:miner_fee_bytes_per_token]
     Enum.filter(txs, fn(tx) ->
-      tx_size_bits = tx
-        |> :erlang.term_to_binary()
-        |> Bits.extract()
-        |> Enum.count()
-      tx_size_bytes = tx_size_bits / 8
+      tx_size_bytes = tx |> :erlang.term_to_binary() |> :erlang.byte_size()
       tx.data.fee >= Float.floor(tx_size_bytes / miners_fee_bytes_per_token)
     end)
   end
 
+  defp filter_transactions_by_block_size(txs, new_block_size_bytes) do
+    {valid_txs_by_block_size, _} =
+      List.foldl(txs, {[], false}, fn(tx, {selected_txs, limit_reached}) ->
```

Comment:
We will go through every transaction even if we've reached the limit, a recursive function that stops when the limit is reached should be better here.